packages/target-tauri/src-tauri/src/state/tray_manager.rs

```diff
@@ -0,0 +1,117 @@
+    menus::tray_menu::create_tray_menu,
+    tray::{build_tray_icon, is_tray_icon_active},
+    DeltaChatAppState,
+};
+
+pub(crate) struct TrayManager {
+    pub(crate) tray: RwLock<Option<TrayIcon>>,
+}
+
+impl TrayManager {
+    pub fn new() -> Self {
+        Self {
+            tray: RwLock::new(None),
+        }
+    }
+
+    /// apply whether to show or hide the tray icon
+    pub async fn apply_wanted_active_state(&self, app: &AppHandle) -> anyhow::Result<()> {
+        let wanted_state = is_tray_icon_active(app)?;
+        let currently_active = { self.tray.read().await.is_some() };
+
+        if currently_active != wanted_state {
+            if wanted_state {
+                let tray = build_tray_icon(app)?;
+                let previous = self.tray.write().await.replace(tray);
+                assert!(previous.is_none());
+                self.update_badge(app).await?;
+            } else {
+                let previous = self.tray.write().await.take();
+                if let Some(tray) = previous {
+                    tray.set_visible(false)?;
+                    let tray_option = app.remove_tray_by_id(tray.id());
+                    drop(tray_option);
+                } else {
+                    unreachable!()
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    pub(crate) async fn update_menu(&self, app: &AppHandle) -> anyhow::Result<()> {
+        let lock = self.tray.read().await;
+        if let Some(tray) = lock.as_ref() {
+            let new_menu = create_tray_menu(app)?;
+            tray.set_menu(Some(new_menu))?;
+        }
+
+        Ok(())
+    }
+
+    async fn update_badge(&self, app: &AppHandle) -> anyhow::Result<()> {
+        let counter = {
+            let dc = app.state::<DeltaChatAppState>();
+            let accounts = dc.deltachat.read().await;
+            let mut counter = 0;
+            for account_id in accounts.get_all() {
+                if let Some(account) = accounts.get_account(account_id) {
+                    if let Ok(mute_state) = account.get_ui_config("ui.is_muted").await {
+                        if mute_state != Some("1".to_owned()) {
+                            // account not muted
+                            counter += account.get_fresh_msgs().await?.len();
+                        }
+                    }
+                } else {
+                    log::warn!("could not read account");
+                }
+            }
+            counter
+        };
+
+        self.update_badge_counter(app, counter).await?;
+
+        Ok(())
+    }
+
+    async fn update_badge_counter(&self, app: &AppHandle, counter: usize) -> anyhow::Result<()> {
+        if cfg!(target_os = "macos") {
+            return Ok(());
+        }
+
+        let lock = self.tray.read().await;
+        if let Some(tray) = lock.as_ref() {
+            let asset = match counter {
+                0 => "images/tray/deltachat.png",
+                _ => "images/tray/deltachat-unread.png",
```

Comment:
The icon is jagged on Windows, do you know what we could do about it?

![image](https://github.com/user-attachments/assets/6da0283a-8a01-4038-8106-6d5aed450a88)

Reply:
maybe use the `.ico` file for windows?

Reply:
I pushed a commit with that makes it use the ice file on windows

Reply:
well then I have no idea, it does not look as broken on my windows machine